src/main/java/com/devcycle/sdk/server/local/api/DVCLocalClient.java

```diff
@@ -119,38 +122,102 @@ public <T> Variable<T> variable(User user, String key, T defaultValue) {
       }
       return defaultVariable;
     }
-    String variableJSON = null;
+
+    double appBuild = Double.NaN;
+    try{
+      appBuild = Double.parseDouble(user.getAppBuild());
+    }catch(Exception e){}
+
+    DVCUser_PB user_pb = DVCUser_PB.newBuilder()
+            .setUserId( user.getUserId())
+            .setEmail(createNullableString(user.getEmail()))
+            .setName(createNullableString(user.getName()))
+            .setLanguage(createNullableString(user.getLanguage()))
+            .setCountry(createNullableString(user.getCountry()))
+            .setAppBuild(createNullableDouble(appBuild))
+            .setAppVersion(createNullableString(user.getAppVersion()))
+            .setCustomData(createNullableCustomData(user.getCustomData()))
+            .setCustomData(createNullableCustomData(user.getPrivateCustomData()))
+            .build();
+
+    VariableType_PB pbVariableType = typeEnumToVariableTypeProtobuf(variableType);
+
+    VariableForUserParams_PB params = VariableForUserParams_PB.newBuilder()
+            .setSdkKey(sdkKey)
+            .setUser(user_pb)
+            .setVariableKey(key)
+            .setVariableType(pbVariableType)
+            .setShouldTrackEvent(true)
+            .build();
+
     try {
-      variableJSON = localBucketing.getVariable(sdkKey, user, key, variableType, true);
-      if (variableJSON == null || variableJSON.isEmpty()) {
+      byte[] paramsBuffer = params.toByteArray();
+      byte[] variableData = localBucketing.getVariableForUserProtobuf(paramsBuffer);
+
+      if (variableData == null || variableData.length == 0) {
         return defaultVariable;
       } else {
-        ObjectMapper mapper = new ObjectMapper();
-        Variable baseVariable = mapper.readValue(variableJSON, Variable.class);
-
-        Variable<T> variable = (Variable<T>) Variable.builder()
-                .key(key)
-                .type(baseVariable.getType())
-                .value(baseVariable.getValue())
-                .defaultValue(defaultValue)
-                .isDefaulted(false)
-                .build();
-        if (variable.getType() != variableType) {
+
+        SDKVariable_PB sdkVariable = SDKVariable_PB.parseFrom(variableData);
+
+        if(sdkVariable.getType() != pbVariableType)
+        {
           System.out.printf("Variable type mismatch, returning default value");
           return defaultVariable;
         }
-        variable.setDefaultValue(defaultValue);
-        variable.setIsDefaulted(false);
+
+        Variable<T> variable;
+        switch(sdkVariable.getType()) {
```

Comment:
similar here, think we can break this off into its own function if we can't make it generic?

Reply:
agreed

Reply:
i broke out some of this stuff into a helper class. The User/Variable classes are in a common package and the protobuf stuff is all isolated to the local package and I didn't want to create cross dependencies between common and local, ideally local should depend on common and common depends on nothing.